### ISSUE TITLE
Composer can not resolve dependency (and hangs forever)

### ISSUE THREAD

**Issue by goetas** [composer/composer#7147]
My `composer.json`:

```json
{
    "name": "my/project",
    "description": "my project",
    "license": "proprietary",
    "require" : {
        "symfony/twig-bundle" : "^3.0",
        "symfony/framework-bundle" : "^2.8"
    }
}

```

Output of `composer diagnose`:
```
$ composer diagnose
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.6.3
PHP version: 7.2.2-3+ubuntu16.04.1+deb.sury.org+1
PHP binary path: /usr/bin/php7.2
```
Using Composer version `1.6.3`.

When I run this command:

```
composer install -vvv
```

I get the following output:

```
$ composer install -vvv
Reading ./composer.json
Loading config file /home/goetas/.composer/config.json
Loading config file /home/goetas/.composer/auth.json
Loading config file ./composer.json
Checked CA file /etc/ssl/certs/ca-certificates.crt: valid
Executing command (/home/goetas/projects/composer-resolve): git branch --no-color --no-abbrev -v
Reading /home/goetas/.composer/composer.json
Loading config file /home/goetas/.composer/config.json
Loading config file /home/goetas/.composer/auth.json
Loading config file /home/goetas/.composer/composer.json
Loading config file /home/goetas/.composer/auth.json
Reading /home/goetas/.composer/auth.json
Running 1.6.3 (2018-01-31 16:28:17) with PHP 7.2.2-3+ubuntu16.04.1+deb.sury.org+1 on Linux / 4.13.0-32-generic
Loading composer repositories with package information
Downloading https://packagist.org/packages.json
Writing /home/goetas/.composer/cache/repo/https---packagist.org/packages.json into cache
Updating dependencies (including require-dev)
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-2013.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-2014.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-2015.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-2016.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-2017.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-2017-04.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-2017-07.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-2017-10.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-2018-01.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-archived.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/p-provider-latest.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$twig-bundle.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$asset.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$http-foundation.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$http-kernel.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$twig-bridge.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$polyfill-php54.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$polyfill-php55.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$polyfill-mbstring.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$polyfill-php70.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-psr$log.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$debug.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$event-dispatcher.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-twig$twig.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-ircmaxell$password-compat.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-paragonie$random-compat.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$config.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$filesystem.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$framework-bundle.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-doctrine$annotations.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-doctrine$cache.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$class-loader.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$dependency-injection.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$routing.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$security-core.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$security-csrf.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$stopwatch.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$templating.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$translation.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-doctrine$lexer.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$polyfill-apcu.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$polyfill-php56.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$polyfill-util.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$finder.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$var-dumper.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$form.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$console.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-doctrine$common.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-symfony$expression-language.json from cache
Reading /home/goetas/.composer/cache/repo/https---packagist.org/provider-phpunit$phpunit.json from cache
Resolving dependencies through SAT
```

Composer is stuck on `Resolving dependencies through SAT` and hangs forever.
Apparently can not resolve the dependencies.

I have created a test project, the code is on https://github.com/goetas/composer-resolve
and the failing TravisCI build is https://travis-ci.org/goetas/composer-resolve/builds/347876156
Inverting the order of the dependencies, works!

```json
{
    "name": "my/project",
    "description": "my project",
    "license": "proprietary",
    "require" : {
        "symfony/framework-bundle" : "^2.8",
        "symfony/twig-bundle" : "^3.0"
    }
}

```

**Comment by Arekusei** [comment]
Having same issue. Earlier composer install works fine. But now when I back to my project composer hang on SAT.
What changes have been made in composer that caused similar problems? Anybody know?
Thanks

**Comment by ladhadha** [comment]
Having same issue. It hangs almost 6 hours, before I kill it.

```
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-yajra$laravel-datatables-html.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-yajra$laravel-datatables-oracle.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-yajra$laravel-datatables-buttons.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-orchestra$testbench.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-fzaninotto$faker.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-orchestra$database.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-symfony$dom-crawler.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-orchestra$testbench-core.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-laravel$laravel.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-webpatser$laravel-uuid.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-tightenco$collect.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$auth.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$broadcasting.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$bus.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$cookie.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$encryption.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$exception.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$hashing.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$log.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$mail.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$notifications.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$pagination.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$queue.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$redis.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$translation.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-illuminate$validation.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-rhumsaa$uuid.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-symfony$dependency-injection.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-symfony$config.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-twig$twig.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$batch.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$cache.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$common.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$http.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$inflection.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$iterator.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$log.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$parser.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-async.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-backoff.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-cache.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-cookie.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-curlauth.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-history.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-log.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-mock.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-oauth.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$service.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$stream.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-md5.json from cache
Reading /home/dharma/.cache/composer/repo/https---packagist.org/provider-guzzle$plugin-error-response.json from cache
Resolving dependencies through SAT
```

```
$ composer diagnose                                                             
Checking composer.json: WARNING
require.league/flysystem : exact version constraints (1.0.9) should be avoided if the package follows semantic versioning
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.6.5
PHP version: 7.2.5
PHP binary path: /usr/bin/php
```

```
{
	"name": "laravel/laravel",
	"description": "The Laravel Framework.",
	"keywords": ["framework", "laravel"],
	"license": "MIT",
	"type": "project",
	"require": {
		"php":">=5.6.4",
        "laravel/framework": "5.4.*",
		"laravelcollective/html": "~5.4",
		"laravel/socialite": "^3.0",
		"cartalyst/sentinel": "2.0.*",
		"cartalyst/tags": "4.0.*",
		"maatwebsite/excel": "~2.1.0",
		"anhskohbo/no-captcha": "^2.1",
		"league/flysystem": "1.0.9",
		"league/glide": "^1.0",
		"baum/baum": "^1.1",
		"intervention/image": "^2.3",
		"laravel/tinker": "^1.0",
		"yajra/laravel-datatables-oracle": "^7.0",
		"larapack/config-writer": "1.*",
		"plank/laravel-mediable": "^2.4",
		"mpociot/teamwork": "~2.0",
		"yajra/laravel-datatables-buttons": "^1.0",
		"spatie/laravel-analytics": "^1.4",
		"spatie/laravel-newsletter": "^3.7",
		"khill/lavacharts": "3.0.*",
		"webpatser/laravel-uuid": "2.*"
	},
	"require-dev": {
		"phpunit/phpunit": "4.7.7",
		"phpspec/phpspec": "2.2.1"
	},
	"autoload": {
		"classmap": [
			"database"
		],
		"psr-4": {
			"App\\": "app/"
		}
	},
	"autoload-dev": {
		"classmap": [
			"tests/TestCase.php"
		]
	},
	"scripts": {
		"post-install-cmd": [
			"php artisan clear-compiled",
			"php artisan optimize"
		],
		"post-update-cmd": [
			"php artisan clear-compiled",
			"php artisan optimize"
		],
		"post-create-project-cmd": [
			"php -r \"copy('.env.example', '.env');\"",
			"php artisan key:generate"
		]
	},
	"config": {
		"preferred-install": "dist"
	}
}
```

**Comment by kosmonowt** [comment]
+1 from me:

this does NOT work (3.3.x):
```
{
    "name": "andosto/eventmanager",
    "type": "october-plugin",
    "description": "Event Manager Plugin",
    "require": {
        "jenssegers/mongodb": "3.3.x",
        "simplesoftwareio/simple-qrcode": "*",
        "tymon/jwt-auth": "0.5.*",
        "barryvdh/laravel-cors": "0.9.2",
        "maatwebsite/excel": "~2.1.0",
        "chumper/zipper": "1.0.x",
        "anam/phantommagick": "dev-master",
        "anam/phantomjs-linux-x86-binary": "dev-master",
        "milon/barcode": "^5.1",
        "softon/sms": "dev-master"
    },
    "minimum-stability": "dev"
}
```

this does work (3.0.x):
```
{
    "name": "andosto/eventmanager",
    "type": "october-plugin",
    "description": "Event Manager Plugin",
    "require": {
        "jenssegers/mongodb": "3.0.x",
        "simplesoftwareio/simple-qrcode": "*",
        "tymon/jwt-auth": "0.5.*",
        "barryvdh/laravel-cors": "0.9.2",
        "maatwebsite/excel": "~2.1.0",
        "chumper/zipper": "1.0.x",
        "anam/phantommagick": "dev-master",
        "anam/phantomjs-linux-x86-binary": "dev-master",
        "milon/barcode": "^5.1",
        "softon/sms": "dev-master"
    },
    "minimum-stability": "dev"
}
```

**Comment by mhmohon** [comment]
Having same issue. does anyone know how to solve this??

**Comment by JonahKlimack** [comment]
Also, same issue.  Been trying to get it to work for a day now.  Anyone?
Can anyone suggest better way to troubleshoot than removing one line at a time?  This is a bit of a slow process.

**Comment by rhildred** [comment]
I hadn't touched composer or this old project for a couple of years. Bringing it up with docker. At one point it complained that I didn't have git. When I installed git still seems to hang.

{
    "name": "rhildred/breakthroughs",
    "description": "https://breakthroughs.ysaas.ca",
    "license": "proprietary",
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/rhildred/Git.php"
        },
        {
            "type": "vcs",
            "url": "https://github.com/rhildred/cloudflare"
        }
    ],

    "authors": [
        {
            "name": "Rich Hildred",
            "email": "rhildred@gmail.com"
        }
    ],
    "minimum-stability": "dev",
    "require": {
        "rhildred/git.php": "dev-master",
        "rhildred/cloudflare": "dev-master",
        "brtriver/dbup": "*"
    }
}


bash-4.4$ composer install -vvv
Reading ./composer.json
Loading config file /.composer/auth.json
Loading config file ./composer.json
Checked CA file /etc/ssl/certs/ca-certificates.crt: valid
Executing command (/app): git branch --no-color --no-abbrev -v
Failed to initialize global composer: Composer could not find the config file: /.composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Reading /app/vendor/composer/installed.json
Running 1.8.3 (2019-01-30 08:31:33) with PHP 7.3.1 on Linux / 4.14.94-89.73.amzn2.x86_64
Loading composer repositories with package information
Downloading https://api.github.com/repos/rhildred/Git.php
Downloading https://api.github.com/repos/rhildred/Git.php/contents/composer.json?ref=master
Downloading https://api.github.com/repos/rhildred/Git.php/commits/master
Downloading https://api.github.com/repos/rhildred/Git.php/tags?per_page=100
Downloading https://api.github.com/repos/rhildred/Git.php/git/refs/heads?per_page=100
Reading composer.json of rhildred/Git.php (v0.1.1-rc)
Downloading https://api.github.com/repos/rhildred/Git.php/contents/composer.json?ref=6145ad374bf47a85f36e11923ea9e533c91a14e0
Skipped tag v0.1.1-rc, no composer file was found
Reading composer.json of rhildred/Git.php (master)
Reading /.composer/cache/repo/github.com/rhildred/Git.php/5e595fd624d101d92a810343964478004fedfc4a from cache
Importing branch master (dev-master)
Reading composer.json of rhildred/Git.php (0.2.0)
Reading /.composer/cache/repo/github.com/rhildred/Git.php/29be29bfafe5efa4bbdcf12d9352204bef9da95b from cache
Importing branch 0.2.0 (0.2.0.x-dev)
Downloading https://api.github.com/repos/rhildred/cloudflare
Downloading https://api.github.com/repos/rhildred/cloudflare/contents/composer.json?ref=master
Downloading https://api.github.com/repos/rhildred/cloudflare/commits/master
Downloading https://api.github.com/repos/rhildred/cloudflare/tags?per_page=100
Downloading https://api.github.com/repos/rhildred/cloudflare/git/refs/heads?per_page=100
Reading composer.json of rhildred/cloudflare (master)
Reading /.composer/cache/repo/github.com/rhildred/cloudflare/6f8e348fb56cb56427440b29b08b0cf0436a5a7d from cache
Importing branch master (dev-master)
Downloading https://repo.packagist.org/packages.json
Writing /.composer/cache/repo/https---repo.packagist.org/packages.json into cache
Updating dependencies (including require-dev)
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-2013.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-2014.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-2015.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-2016.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-2017.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-2018.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-2018-04.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-2018-07.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-2018-10.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-2019-01.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/p-provider-archived.json from cache
Downloading http://repo.packagist.org/p/provider-latest%240f1d7b14fad22093a4b82aca86616eee2a59e124ca7f8d90636acf5ad696b558.json
Writing /.composer/cache/repo/https---repo.packagist.org/p-provider-latest.json into cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-brtriver$dbup.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$console.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$event-dispatcher.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$finder.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-phake$phake.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-hamcrest$hamcrest-php.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-phpunit$phpunit.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-code-coverage.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-file-iterator.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-text-template.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-timer.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-phpunit$phpunit-mock-objects.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-sebastian$comparator.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-sebastian$diff.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-sebastian$environment.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-sebastian$exporter.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-sebastian$version.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$yaml.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-token-stream.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-ocramius$instantiator.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-doctrine$instantiator.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-sebastian$recursion-context.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-ctype.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-ocramius$lazy-map.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-piece$stagehand-testrunner.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-piece$stagehand-alterationmonitor.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-piece$stagehand-componentfactory.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$class-loader.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$config.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$dependency-injection.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$process.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-apcu.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$filesystem.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$debug.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-mbstring.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$contracts.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-php73.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-psr$container.json from cache
Reading /.composer/cache/repo/https---repo.packagist.org/provider-psr$log.json from cache

**Comment by rhildred** [comment]
It finally finished:

Resolving dependencies through SAT
Looking at all rules.
Something's changed, looking at all rules again (pass #1)
Dependency resolution completed in 0.253 seconds
Analyzed 5898 packages to resolve dependencies
Analyzed 835643 rules to resolve dependencies
Nothing to install or update
Reading /app/vendor/composer/installed.json
Writing lock file
Generating autoload files

**Comment by jam1e** [comment]
I was having the same issue. After running the following command, things seem to be back to normal:

`composer clear-cache`

**Comment by styzzz** [comment]
Same as Jam1e - running this before update worked for me: 

`composer clear-cache
`

**Comment by mo3000** [comment]
similar issue, composer hangs for ever and clear-cache doesn't work.

> Resolving dependencies through SAT
Looking at all rules.
Something's changed, looking at all rules again (pass #462)

**Comment by Seldaek** [comment]
Closing as most likely fixed by Composer 2.0, please open a new issue if you still experience this with the latest version.

**Comment by foolishnoob** [comment]
I have resolved this problem by upgrade composer v1.x to v2.x use :
```
composer self-update # upgrade composer
composer self-update --update-keys # this step need us open broswer https://composer.github.io/pubkeys.html and paste Dev / Snapshot Public Key and Tags Public Key
composer diagnose # diagnose ,if required ,take ON disable-tls and take OFF secure-http option
```
then use :
```
composer install --dev -vvv
```
